### ISSUE TITLE
Enable opening pop-up windows for authentication

### ISSUE THREAD

**Issue by mstroeck** [djyde/WebShell#21]
I have not been able to get any app to work that uses, for example, my Google account for login. This makes WebShell pretty much unusable for me, and I guess for quite a lot of people

**Comment by djyde** [comment]
I don't quietly understand what you mean `unusable`. After you signing in WebShell-based app, cookies and session would be saved.

**Comment by djyde** [comment]
Oh I see your point. I will think about it.

**Comment by mdamjanic7** [comment]
I'm having the same problem. Tried to setup Asana, but I can't log in.

**Comment by djyde** [comment]
@wdg have any ideas about it?

**Comment by 0xWDG** [comment]
Yes, i will take a look for injecting popups as well as a web hook.
<s>The only challenge is to return the cookies/session/data, since web views can't be linked.</s> since we can use NSUserDefaults sync this is not a problem

<s>i know a proper solution to grab the cookies, and sessions, but to inject it before the web view will load is the only challenge.</s>, After some research this is not a problem if we'll sync.

@djyde we'll need to use some command line arguments, i've made a (little) proof-of-concept

---

## Research part:

Swift code

```
override func viewDidLoad() {
    ....

    let args = [String](Process.arguments)
    print(args)
}
```

Command line

```
wdg@WebShell$ (snip)/WebShell.app/Contents/MacOS/WebShell -width=12 -height=20 -init=0,5
```

Swift printing:

```
["(snip)/WebShell.app/Contents/MacOS/WebShell", "-width=12", "-height=20", "-init=0,5"]
```

---

Analyse of cookies with:

```
    let cookieList : NSHTTPCookieStorage = NSHTTPCookieStorage.sharedHTTPCookieStorage()
    for cookie in cookieList.cookies! as [NSHTTPCookie]{
        NSLog("%@", cookie)
    }
```

returns all the saved cookies however we'll need to use `NSUserDefaults.standardUserDefaults().synchronize()` for Synchronize for further use when opening a new window, and closing a window. to be safe that we'll not destroy/missing a cookie when opening a popup

This all is possible, and then there is a big need to fix #23 since a lot of websites still uses iframes, but i'm not yet know a solution for it.

**Comment by djyde** [comment]
It seems a thorny problem.
